samples/client/petstore/go-experimental/go-petstore/api/openapi.yaml

```diff
@@ -1422,14 +1422,17 @@ components:
       allOf:
       - $ref: '#/components/schemas/Animal'
       - $ref: '#/components/schemas/Dog_allOf'
+      type: object
```

Comment:
In JSON schema, allOf means a compliant validator must validate the payload against all of the nested schemas under `allOf`. Since Animal is already specified as a `type: object`, it's redundant and verbose to add `type: object` here.